plugin/vscode/performance-writer.js

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2020  Matthew Glazar
+// See end of file for extended copyright information.
+
+const fs = require('fs');
+const path = require("path");
+const process = require("process");
+
+class PerformanceWriter {
+    
+    constructor (extensionContext) {
+        this._performanceTxtPath = path.join(process.cwd(), "test_performance.txt");
```

Comment:
Blocking: We should pick a better file name. Also, cwd is probably the wrong place to put our files.

Does VS Code have a place for extensions to store log files or similar data? If so, we should use that place. If not, we should discuss a good home.

Reply:
There is a logging schema from vscode for extension logging files. The schema is as follows: For each new process of vscode, (i.e. you open a new vscode process, this is NOT the same as opening a new window of vscode from an existing process) it adds a folder with a timestamp on to **%AppData%\Code\logs\_timestamped folder_** in windows and **~/.config/Code/logs/_timestamped folder_/** (Untested on linux, saw from [overstack post](https://stackoverflow.com/questions/46373762/where-are-visual-studio-code-log-files)) in linux. For each new workspace that the process has, it spawns a %AppData%\Code\logs\exthost# where # represents a given workspace. Inside of this workspace folder is where vscode wants you to store the extension log data. It automatically generates a URI for you via [extensionContext.logUri](https://code.visualstudio.com/api/references/vscode-api#ExtensionContext).path. This folder does not automatically exist, you must create it. 

Implications: If we were to use vscode's extension schema, your initial example of keeping track of the files using qljs will still work but it may look abit weird in the sense that the initial index of the file and it's associated file name will be in disparate process folders. How would you like to proceed with the design of how you want to store the index and it's linking file name?

Reply:
> Implications: If we were to use vscode's extension schema, your initial example of keeping track of the files using qljs will still work but it may look abit weird in the sense that the initial index of the file and it's associated file name will be in disparate process folders.

This sounds fine to me. Not at all weird.

> How would you like to proceed with the design of how you want to store the index and it's linking file name?

What's the deletion policy for these folders? Does VS Code delete them automatically? Do we need to delete them ourselves?